browser_use/browser/context.py

```diff
@@ -442,14 +474,22 @@ async def _create_context(self, browser: PlaywrightBrowser):
 		"""Creates a new browser context with anti-detection measures and loads cookies if available."""
 		if self.browser.config.cdp_url and len(browser.contexts) > 0 and not self.config.force_new_context:
 			context = browser.contexts[0]
+			# For existing contexts, we need to set the viewport size manually
+			if context.pages and not self.browser.config.headless:
+				await self._set_viewport_size_for_page(context.pages[0])
 		elif self.browser.config.browser_binary_path and len(browser.contexts) > 0 and not self.config.force_new_context:
 			# Connect to existing Chrome instance instead of creating new one
 			context = browser.contexts[0]
+			# For existing contexts, we need to set the viewport size manually
+			if context.pages and not self.browser.config.headless:
+				await self._set_viewport_size_for_page(context.pages[0])
 		else:
 			kwargs = {}
-			if self.browser.config.headless:
-				kwargs['viewport'] = self.config.browser_window_size.model_dump()
-				kwargs['no_viewport'] = False
+			# Set viewport for both headless and non-headless modes
+			kwargs['viewport'] = self.config.browser_window_size.model_dump()
+			# Important: set no_viewport to False to ensure the viewport size is applied
+			kwargs['no_viewport'] = False
```

Comment:
viewport size should not be set in headful mode, it should default to the window size